packages/headless/src/controllers/triggers/headless-redirection-trigger.ts

```diff
@@ -0,0 +1,81 @@
+import {Engine} from '../../app/headless-engine';
+import {
+  ConfigurationSection,
+  RedirectionSection,
+} from '../../state/state-sections';
+import {configuration, redirection} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {logTriggerRedirect} from '../../features/redirection/redirection-analytics-actions';
+
+/**
+ * The `RedirectionTrigger` controller handles redirection actions.
+ */
+export interface RedirectionTrigger extends Controller {
+  /**
+   * the state of the `RedirectionTrigger` controller.
+   */
+  state: RedirectionTriggerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `RedirectionTrigger` controller.
+ */
+export interface RedirectionTriggerState {
+  /**
+   * The url used for the redirection.
+   */
+  redirectTo: string | null;
```

Comment:
I think it would be possible to remove `null` from the type, and always initialize as a string.

Reply:
I initialized it to null for the case where a redirection trigger is received and saved to the engine's state but a redirection-trigger controller has not been initialized yet. Upon its initialization, it would need to immediately deal with that trigger and log it. If we immediately initialize it to the value in state then we would lose that functionality.